src/Relation/Binary/Lattice/Definitions.agda

```diff
@@ -0,0 +1,30 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions for order-theoretic lattices 
+------------------------------------------------------------------------
+
+-- The contents of this module should be accessed via `Relation.Binary.Lattice`.
+
+{-# OPTIONS --without-K --safe #-}
+
+module Relation.Binary.Lattice.Definitions where
+
+open import Algebra.Core
+open import Data.Product using (_×_; _,_)
+open import Function.Base using (flip)
+open import Relation.Binary
+
+------------------------------------------------------------------------
+-- Relationships between orders and operators
+
+Supremum : ∀ {a ℓ} {A : Set a} → Rel A ℓ → Op₂ A → Set _
```

Comment:
We use `variable`s to clean a few of these types up?

Reply:
@MatthewDaggitt  Not sure but is this right?

`private`
  `variable`
    `a ℓ : Level
    A : Set a`

`Supremum : Rel A ℓ → Op₂ A → Set _`
`Supremum _≤_ _∨_ =  ∀ x y → x ≤ (x ∨ y) × y ≤ (x ∨ y) × ∀ z → x ≤ z → y ≤ z → (x ∨ y) ≤ z`

Reply:
Apologies for the delay. Yes that looks good :+1:

Reply:
Thanks. Done